benchmark-timed-schedule.yaml

```diff
@@ -0,0 +1,309 @@
+              title_link: $ATC_EXTERNAL_URL/builds/$BUILD_ID
+      on_success:
+        put: slack-alert
+        params:
+          channel: '#((slack_channel_name))'
+          attachments:
+            - pretext: Survey Benchmark Deploy Passed
+              color: good
+              title: Concourse Build $BUILD_ID
+              title_link: $ATC_EXTERNAL_URL/builds/$BUILD_ID
+
+
+##############################################################################
+# Destroy Cluster
+##############################################################################
+
+  - name: destroy-load-generator
+    serial_groups: [deploy-survey-runner-benchmark]
+    plan:
+    - get: eq-terraform-load-generator
+    - get: eq-survey-runner-benchmark
+      passed: [build-eq-survey-runner-benchmark, deploy-survey-runner-benchmark]
+    - get: scheduled-benchmark-trigger
+      passed: [build-eq-survey-runner-benchmark, deploy-survey-runner-benchmark]
+      trigger: true
+    - task: Destroy Load Generator
+      config:
+        platform: linux
+        image_resource:
+          type: docker-image
+          source:
+            repository: google/cloud-sdk
+        params:
+          TF_VAR_k8s_min_master_version: ((k8s_min_master_version))
+          TF_VAR_gcp_folder_id: ((gcp_folder_id))
+          TF_VAR_gcp_billing_account: ((gcp_billing_account))
+          SERVICE_ACCOUNT_JSON: ((gcp.service_account_json))
+          GOOGLE_APPLICATION_CREDENTIALS: /root/gcloud-service-key.json
+        inputs:
+          - name: eq-terraform-load-generator
+        run:
+          path: bash
+          args:
+            - -exc
+            - |
+              apt-get install -y unzip
+
+              git clone https://github.com/kamatama41/tfenv.git ~/.tfenv && \
+              ln -s /root/.tfenv/bin/* /usr/local/bin
+
+              cat >$GOOGLE_APPLICATION_CREDENTIALS <<EOL
+              $SERVICE_ACCOUNT_JSON
+              EOL
+
+              gcloud auth activate-service-account --key-file $GOOGLE_APPLICATION_CREDENTIALS
+
+              cd eq-terraform-load-generator
+
+              tfenv install
+
+              terraform init --upgrade --backend-config prefix=((project_name)) -var "project_name=((project_name))" --backend-config bucket=((load_generator_tf_state_bucket))
+
+              # Do not delete the project or bucket.
+              terraform state rm google_project.project
+              terraform state rm google_storage_bucket.benchmark-output-storage
+
+              # Destroy infrastructure
+              terraform destroy --auto-approve -var "project_name=((project_name))"
+
+              PROJECT_ID=($(gcloud projects list --filter=name="((project_name))" --format="value(PROJECT_ID)"))
+
+              # Import project and bucket resouce back into the state
+              terraform import google_project.project "$PROJECT_ID"
+              terraform import google_storage_bucket.benchmark-output-storage ${PROJECT_ID}-benchmark-outputs
```

Comment:
Have you tried using a lifecycle rule rather than editing the state?

```
lifecycle {
  prevent_destroy = true
}
```

Reply:
As discussed, this just prevents commands from destroying the resource with this specified and not what we are after which is to ignore it when destroying.